sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -280,3 +283,210 @@ async fn test_upload_blob_overwrite(ctx: TestContext) -> Result<(), Box<dyn Erro
     container_client.delete_container(None).await?;
     Ok(())
 }
+
+#[recorded::test]
+async fn test_put_block_list(ctx: TestContext) -> Result<(), Box<dyn Error>> {
+    // Recording Setup
+    let recording = ctx.recording();
+    let (options, endpoint) = recorded_test_setup(recording, BlobClientOptions::default()).await;
+    let container_name = recording
+        .random_string::<17>(Some("container"))
+        .to_ascii_lowercase();
+    let blob_name = recording
+        .random_string::<12>(Some("blob"))
+        .to_ascii_lowercase();
+
+    // Act
+    let container_client = ContainerClient::new(
+        &endpoint,
+        container_name.clone(),
+        recording.credential(),
+        Some(options.clone()),
+    )?;
+    container_client.create_container(None).await?;
+
+    let blob_client = BlobClient::new(
+        &endpoint,
+        container_name,
+        blob_name,
+        recording.credential(),
+        Some(options),
+    )?;
+
+    let block_1 = b"AAA";
+    let block_2 = b"BBB";
+    let block_3 = b"CCC";
+
+    blob_client
+        .stage_block(
+            "1",
+            i64::try_from(block_1.len())?,
+            RequestContent::from(block_1.to_vec()),
+            Some(BlobBlockBlobClientStageBlockOptions::default()),
+        )
+        .await?;
+
+    blob_client
+        .stage_block(
+            "2",
+            i64::try_from(block_2.len())?,
+            RequestContent::from(block_2.to_vec()),
+            Some(BlobBlockBlobClientStageBlockOptions::default()),
+        )
+        .await?;
+    blob_client
+        .stage_block(
+            "3",
+            i64::try_from(block_3.len())?,
+            RequestContent::from(block_3.to_vec()),
+            Some(BlobBlockBlobClientStageBlockOptions::default()),
+        )
+        .await?;
+
+    let latest_blocks: Vec<String> = vec![
+        base64::encode("1"),
+        base64::encode("2"),
+        base64::encode("3"),
+    ];
+    let block_lookup_list = BlockLookupList {
+        committed: None,
+        latest: Some(latest_blocks),
+        uncommitted: None,
+    };
+
+    let request_content = RequestContent::try_from(block_lookup_list)?;
+
+    blob_client
+        .commit_block_list(
+            request_content,
+            Some(BlobBlockBlobClientCommitBlockListOptions::default()),
+        )
+        .await?;
+
+    let response = blob_client
+        .download_blob(Some(BlobBlobClientDownloadOptions::default()))
+        .await?;
+
+    // Assert
+    let (status_code, headers, response_body) = response.deconstruct();
+    assert!(status_code.is_success());
+    assert_eq!(
+        "9",
+        headers.get_str(&HeaderName::from_static("content-length"))?
+    );
+    assert_eq!(
+        Bytes::from_static(b"AAABBBCCC"),
+        response_body.collect().await?
+    );
+
+    container_client.delete_container(None).await?;
+    Ok(())
+}
+
+// Currently blocked by generated code, uncomment when we can consume newest definition
```

Comment:
This definitely feels like an instance where we can test `get_block_list` (when available) inside the same test above. Maybe rename it to `test_block_list` and put everything in one?

Reply:
Sounds good, I'll just remove both the def of the method & tests and then lump it in once we get past this PR.